### PR TITLE
README Changes and UI Improvements

### PR DESCRIPTION
**Pull request by omairqazi**
Added screenshot and instructions to run Randomizer to README.

### CODE REVIEW

**Comment by scottmc** [comment]
As part of this cleanup please also delete the Randomizer.rsrc and Randomizer.proj files as they are no longer needed.  Also verify that the makefile will still build the program.

File: README.md [hunk 1]
```diff
@@ -1,8 +1,18 @@
 Randomizer
 ==========
 
+![screenshot](screenshot.png)
+
 Random symbol sequence (e.g. password) generator for Haiku
 
+Instructions:
```
Comment (humdingerb) [inline README.md]: Maybe call it "Build instructions:".

File: README.md [hunk 2]
```diff
@@ -1,8 +1,18 @@
...
+  * Clone the repo and `cd` to the folder where you cloned it and run the following:
```
Comment (humdingerb) [inline README.md]: I think ou can be even more specific, as that folder gets cloned and its name therefore known... So just: "Clone the repo and run the following:"
Then in start with "cd Randomizer/src".

File: README.md [hunk 3]
```diff
@@ -1,8 +1,18 @@
...
+  ```
+  cd src
+  make
+  ./Randomizer
+  ```
+
```
Comment (humdingerb) [inline README.md:15]: You can remove everything below. It's all outdated or has no informative contents.

Comment (omairqazi) [reply README.md]: Removed the project webpage and changed the requirements.

**Comment by omairqazi** [comment]
> As part of this cleanup please also delete the Randomizer.rsrc and Randomizer.proj files as they are no longer needed. Also verify that the makefile will still build the program.

Done :) https://github.com/HaikuArchives/Randomizer/pull/2 & https://github.com/HaikuArchives/Randomizer/pull/3

**Comment by omairqazi** [comment]
This PR now consists commits of UI Improvements and updated screenshot and README.

PRs https://github.com/HaikuArchives/Randomizer/pull/2 and https://github.com/HaikuArchives/Randomizer/pull/3 has been merged into this singe PR thereby closing them. :)

**Comment by mmuman** [comment]
Hmm, both BBox (and the child class RandoBox) and BTextControl already have a `label` parameter on the constructor, you shouldn't need to add a separate StringView.

File: src/AppWindow.cpp [hunk 4]
```diff
@@ -28,55 +29,75 @@ AppWindow::AppWindow(BRect frame)
...
-	PassOut = new RandoTextView(BRect(5,25,200,40), "PassOut",
-		BRect(0,0,200,30), be_bold_font, NULL, B_FOLLOW_LEFT|B_FOLLOW_TOP,
+	// Result Label:
+	BStringView *result = new BStringView(BRect(100,30,11,11),"resultlabel","RESULT");
+	result->ResizeToPreferred();
+	MainView->AddChild(result);
+	
+	// Result:
+	PassOut = new RandoTextView(BRect(5,50,245,80), "PassOut",
+		BRect(5,0,245,60), be_bold_font, NULL, B_FOLLOW_LEFT|B_FOLLOW_TOP,
```
Comment (owenca) [inline src/AppWindow.cpp:39]: It's better to change `RandoTextView` to a `BTextControl` if we want to add a label to the `PassOut` box, but then the window would be too wide. I suggest that we hold off this change until after the layout of the window is redesigned.

File: src/AppWindow.cpp [hunk 5]
```diff
@@ -28,55 +29,75 @@ AppWindow::AppWindow(BRect frame)
...
-			
-	ParamsBox = new RandoBox(BRect(5,50,200,146), "ParamsBox", "", B_FOLLOW_LEFT|B_FOLLOW_TOP,
+	
+	// Parameters Label:
+	BStringView *param = new BStringView(BRect(90,90,11,11),"paramlabel","Parameters:");
+	param->ResizeToPreferred();
+	MainView->AddChild(param);
+	
+	// Parameters Box:
+	ParamsBox = new RandoBox(BRect(5,100,245,210), "ParamsBox", "", B_FOLLOW_LEFT|B_FOLLOW_TOP,
```
Comment (owenca) [inline src/AppWindow.cpp]: Like @mmuman said, `RandoBox` already has a Label argument, the null string (`""`) after `"ParamsBox"`. However, the layout of the window/view must be changed in order for the label to be visible.

File: src/AppWindow.cpp [hunk 6]
```diff
@@ -28,55 +29,75 @@ AppWindow::AppWindow(BRect frame)
...
 		B_WILL_DRAW|B_NAVIGABLE, B_FANCY_BORDER);
 	MainView->AddChild(ParamsBox);
 	
-	PassLength = new BTextControl(BRect(5,5,189,40), "PassLength", "Password length", NULL,
+	// Length Label:
+	BStringView *length = new BStringView(BRect(60,15,100,40),"lengthlabel","Length:");
+	ParamsBox->AddChild(length);
+	
+	// Length TextBox:
+	PassLength = new BTextControl(BRect(70,15,140,40), "PassLength", "Password length", NULL,
```
Comment (owenca) [inline src/AppWindow.cpp]: The label `"Password length"` is already passed to `BTextControl`, but there is a bug that prevents it from visible, though. Can you find it?

**Comment by owenca** [comment]
Let's remove the `ChangeLog` file, which is redundant as we can run `git log` to see the history.

**Comment by omairqazi** [comment]
> Let's remove the `ChangeLog` file, which is redundant as we can run `git log` to see the history.

Done! Made 2 new commits.

File: src/AppWindow.cpp (hunk 6 above)
Comment (omairqazi) [reply src/AppWindow.cpp]: That's a problem with `BTextControl` I think. I tried changing the label to `"123"` in the `BTextControl` for `Custom` but it didn't appear. Note, label appears with `BCheckBox` but not with `BTextControl`.

**Comment by omairqazi** [comment]
Screenshot has been updated to show the new `ParamsBox` label (commit- https://github.com/HaikuArchives/Randomizer/pull/1/commits/ce064507b7583fbfcae6f65b0c7819c51f292f00).

File: src/AppWindow.cpp [hunk 7]
```diff
@@ -28,55 +29,75 @@ AppWindow::AppWindow(BRect frame)
...
 		new BMessage(SEQ_LEN_MSG), B_FOLLOW_LEFT|B_FOLLOW_TOP, B_NAVIGABLE);
```
Comment (owenca) [inline src/AppWindow.cpp]: The bug is that the B_WILL_DRAW flag is not included. If you bitwise-or it with the last argument, the label will appear. :)

File: src/AppWindow.cpp (hunk 6 above)
Comment (owenca) [reply src/AppWindow.cpp]: Please see my comments below.

File: src/AppWindow.cpp (hunk 7 above)
Comment (omairqazi) [reply src/AppWindow.cpp]: Thanks ! Done :) commit https://github.com/HaikuArchives/Randomizer/pull/1/commits/fd485d3eef51150cc25c956a5247109031c410ba

**Comment by omairqazi** [comment]
Screenshot has been updated to show the `Password length` label (commit- https://github.com/HaikuArchives/Randomizer/pull/1/commits/18df7432817ceec7d3d364c2f19664e11aff94e4).

**Comment by scottmc** [comment]
So if this one ready for merge now? @mmuman ?

**Comment by humdingerb** [comment]
@janus2 work has superseded the GUI part of this PR.

If this was a GCI task that's still active, I suggest taking a new screenshot, describe the options and the new copy-to-clipboard shortcut/icon and remove the build instructions (they are obvious to anyone intrested in building from source), system requirements, history and license.

**Comment by humdingerb** [comment]
As stated, all superseded by other commits.

**Pull request closed without merging**